transformer_engine/pytorch/csrc/extensions.h

```diff
@@ -441,3 +441,21 @@ void multi_tensor_sgd_cuda(int chunk_size, at::Tensor noop_flag,
                            bool wd_after_momentum, float scale);
 
 #endif  // TRANSFORMER_ENGINE_PYTORCH_CSRC_EXTENSIONS_H_
+
+/***************************************************************************************************
+ * Support memory efficient cross entropy for Megatron-LM
+ **************************************************************************************************/
+
+at::Tensor cross_entropy_forward_sum_exp(const at::Tensor &vocab_parallel_logits_ptr,
+                                         const at::Tensor &logits_max_ptr);
+
+at::Tensor cross_entropy_fwd_mean_log(const at::Tensor &vocab_parallel_logits_ptr,
+                                      const at::Tensor &logits_max_ptr,
+                                      const at::Tensor &sum_exp_logits_ptr);
+
+at::Tensor cross_entropy_bwd(const at::Tensor &grad_output_ptr,
+                             const at::Tensor &input_ptr,  //vocab_parallel_logits_ptr
+                             const at::Tensor &target_mask_ptr,
+                             const at::Tensor &masked_target_1d_ptr,
+                             const at::Tensor &logits_max_ptr, const at::Tensor &sum_exp_logits_ptr,
```

Comment:
nit, for better readability
```suggestion
                             const at::Tensor &logits_max_ptr, 
                             const at::Tensor &sum_exp_logits_ptr,
```

Reply:
Done,  THX @ptrendx